### PR TITLE
#23612 delete primary goal from experiment

### LINKED ISSUE #23612: [UI] Delete Goal
### Is your feature request related to a problem? Please describe.
Add three dots menu to the card of the selected goal.

### Describe the solution you'd like
Show an option of removing the goal

### Acceptance criteria
- Add a Goal
- With the menu of 3 dots, remove it
- If the card does not have a goal, the title does not show the green check.

```
@DELETE
@Path("/{experimentId}/goals/primary")

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Add an icon to delete the goal
* Show a confirmation to delete

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
N/A

### Screenshots
Original
Screenshot 2023-01-11 at 2 07 31 PM 

Updates:
Screenshot 2023-01-11 at 2 07 36 PM
 
Screenshot 2023-01-11 at 2 07 41 PM

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.html [hunk 1]
```diff
@@ -21,9 +21,18 @@ <h2 class="flex align-items-center gap-1 uppercase">
...
+                                <h3 data-testId="goal-label">
+                                    {{ goalTypeMap[vm.goals.primary.type].label | dm }}
+                                </h3>
+                                <dot-icon
+                                    class="delete cursor-pointer"
+                                    (click)="deleteGoal($event, 'primary')"
+                                    data-testId="goal-delete-button"
+                                    name="delete"
+                                    size="16"
+                                ></dot-icon>
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.html:34]: We should use an primeng icon button here, not just the icons which is just an `<i>` tag.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts [hunk 2]
```diff
@@ -65,6 +74,23 @@ export class DotExperimentsConfigurationGoalsComponent {
         this.dotExperimentsConfigurationStore.openSidebar(ExperimentSteps.GOAL);
     }
 
+    /**
+     * Delete goal
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts:78]:
```suggestion
     * Show confirmation dialog to allow the user confirmation to delete a goal
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts [hunk 3]
```diff
@@ -273,6 +274,32 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
         }
     );
 
+    readonly deleteGoal = this.effect((goalLevel$: Observable<GoalsLevels>) => {
+        return goalLevel$.pipe(
+            withLatestFrom(this.state$),
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts:279]: Again... send the experiment info from the component.

File: core-web/libs/block-editor/src/lib/extensions/image-tabview-form/components/dot-image-search/store/dot-image-search.store.ts [hunk 4]
```diff
@@ -33,51 +34,43 @@ const defaultState: DotImageSearchState = {
...
     });
-
     readonly updateSearch = this.updater<string>((state, search) => {
         return {
             ...state,
             search
         };
     });
-
     // Effects
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/image-tabview-form/components/dot-image-search/store/dot-image-search.store.ts:74]: This line remove are not good.

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts (hunk 2 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts:78]: Done

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts (hunk 3 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts:279]: Fixed.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts [hunk 5]
```diff
@@ -65,6 +74,26 @@ export class DotExperimentsConfigurationGoalsComponent {
...
 
+    /**
+     * Show confirmation dialog to allow the user confirmation to delete a goal
+     * @param event
+     * @param {GoalsLevels} goalLevel
+     * @param {string} experimentId
+     * @returns void
+     * @memberof DotExperimentsConfigurationGoalsComponent
+     */
+    deleteGoal(event, goalLevel: GoalsLevels, experimentId: string) {
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.ts:85]: `event: Event` add type

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/services/dot-experiments.service.ts [hunk 6]
```diff
@@ -139,4 +139,19 @@ export class DotExperimentsService {
             .patch<DotCMSResponse<DotExperiment>>(`${API_ENDPOINT}/${experimentId}`, { goals })
             .pipe(pluck('entity'));
     }
+
+    /**
+     * Delete a goal of an experiment
+     * @returns Observable<DotExperiment>
+     * @memberof DotExperimentsService
+     * @param experimentId
+     * @param goalType
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/services/dot-experiments.service.ts:148]: missing type

File: core-web/libs/block-editor/src/lib/extensions/image-tabview-form/components/dot-image-search/store/dot-image-search.store.ts [hunk 7]
```diff
@@ -15,6 +14,8 @@ import {
 } from '@dotcms/block-editor';
 import { DotCMSContentlet } from '@dotcms/dotcms-models';
 
+const DEFAULT_LANG_ID = 1;
```
Comment (hmoreras) [inline core-web/libs/block-editor/src/lib/extensions/image-tabview-form/components/dot-image-search/store/dot-image-search.store.ts:17]: @rjvelazco checkout this change, it was breaking Arcadio's environment.

**Review by hmoreras — APPROVED** [review]

**Review by alfredo-dotcms — APPROVED** [review]

**Pull request merged**